src/main/java/ortus/boxlang/web/bifs/HtmlParse.java

```diff
@@ -0,0 +1,106 @@
+	 * Some of the most common methods you can use on the Document object include:
+	 * <ul>
+	 * <li><code>title()</code> – Get the contents of the &lt;title&gt; tag.</li>
+	 * <li><code>select(selector)</code> – Find elements using CSS selectors.</li>
+	 * <li><code>text()</code> – Get the combined text of the entire document.</li>
+	 * <li><code>outerHtml()</code> – Get the HTML of the entire document.</li>
+	 * <li><code>body()</code> – Get the &lt;body&gt; element.</li>
+	 * <li><code>head()</code> – Get the &lt;head&gt; element.</li>
+	 * <li><code>getElementById(id)</code> – Find an element by its ID attribute.</li>
+	 * <li><code>getElementsByTag(tagName)</code> – Get all elements with the given tag.</li>
+	 * <li><code>getElementsByClass(className)</code> – Get all elements with the given class.</li>
+	 * <li><code>getElementsByAttribute(attrName)</code> – Get elements that have the specified attribute.</li>
+	 * <li><code>html()</code> – Get the inner HTML of the document body.</li>
+	 * <li><code>createElement(tagName)</code> – Create a new element with the given tag.</li>
+	 * <li><code>charset()</code> – Get the document's character set.</li>
+	 * <li><code>location()</code> – Get the URL from which the document was loaded.</li>
+	 * <li><code>normalise()</code> – Normalize the document structure (e.g., ensures head/body exist).</li>
+	 * </ul>
+	 *
+	 * <h2>Usage</h2>
+	 *
+	 * <pre>
+	 * // Parse a simple HTML string
+	 * htmlContent = "<html><head><title>My Page</title></head><body><h1>Hello World</h1></body></html>";
+	 * doc = htmlParse( htmlContent );
+	 * title = doc.title(); // Returns "My Page"
+	 *
+	 * // Extract text content from HTML
+	 * htmlContent = "<div class='content'><p>First paragraph</p><p>Second paragraph</p></div>";
+	 * doc = htmlParse( htmlContent );
+	 * text = doc.text(); // Returns "First paragraph Second paragraph"
+	 *
+	 * // Select elements using CSS selectors
+	 * htmlContent = "<ul><li class='item'>Item 1</li><li class='item'>Item 2</li></ul>";
+	 * doc = htmlParse( htmlContent );
+	 * items = doc.select( ".item" ); // Returns elements with class 'item'
+	 *
+	 * // Find elements by ID and modify content
+	 * htmlContent = "<div id='main'><p>Original content</p></div>";
+	 * doc = htmlParse( htmlContent );
+	 * mainDiv = doc.getElementById( "main" );
+	 * // You can now work with the mainDiv element
+	 * </pre>
+	 *
+	 * @param context   The context in which the BIF is being invoked.
+	 * @param arguments Argument scope for the BIF.
+	 *
+	 * @argument.html A HTML string to be parsed.
+	 *
+	 * @return A Jsoup Document object representing the parsed HTML content.
+	 */
+	public Object _invoke( IBoxContext context, ArgumentsScope arguments ) {
+		var target = arguments.getAsString( KeyDictionary.html );
+
+		if ( target == null || target.isEmpty() ) {
+			return Jsoup.parse( "" );
+		}
+
+		return Jsoup.parse( target );
```

Comment:
I have mixed feelings on returning the raw DOM document object here - as adding this BIF means that we have to document and support the usage of it.  

I might suggest a wrapping Type ( e.g. `HTML` ) for the Document Object so that we can expose member methods for documentation and even add shortcuts like `toXML()`.  We can add the dereference and invoke to the Type so that methods pass through if they are not found in the type - like we do with `DateTime` to `ZonedDateTime`.

This might also be helpful in adding more fluent search methods to parsed HTML content.   I don't know that this is a blocker for this PR, but it feels like we could make this return value a bit more "native"